### ISSUE TITLE
Not utilizable by Soundflowerbed.

### ISSUE THREAD

**Issue by ShikiSuen** [ExistentialAudio/BlackHole#328]
The Soundflowerbed routing app still runs on macOS Big Sur but cannot work with BlackHole driver.

Soundflowerbed is the only reason I have to install Soundflower.

**Comment by devinroth** [comment]
I'm not going to touch this with a 10 foot pole. Sorry. 
I'm pretty sure Soundflowerbed is no longer supported.

**Comment by ShikiSuen** [comment]
@devinroth May I confirm whether the device name of BlackHole will be changed in the future?
If not, I think the only approach is to recompile SoundflowerBed to let it use BlackHole driver instead.

**Comment by devinroth** [comment]
No plans on that. BlackHole 2ch and BlackHole 16ch. Any subsequent versions will have the same names / UIDs.

**Comment by ShikiSuen** [comment]
@devinroth Many thanks. You guys already did this help.
Now the rest issue is to recompile SoundflowerBed to let it use BlackHole drivers instead.

**Comment by ShikiSuen** [comment]
@devinroth Could you please tell me the "AudioDeviceID" of both BlackHole 2ch and 16ch?
I tried using BBEdit to perform massive search of the string "AudioDeviceID" through the entire BlackHole repo but no search results returned.

The BlackHoleBed is successfully compiled on my side but it has to use AudioDeviceID to identify whether BlackHoleBed has been installed:
<img width="511" alt="image" src="https://user-images.githubusercontent.com/3164826/109592191-357f4500-7b52-11eb-9794-3abcd291d087.png">

**Comment by ShikiSuen** [comment]
Also, where are BlackHoleBed drivers located? I cannot find their kexts.

**Comment by ShikiSuen** [comment]
I assume that "BlackHole2ch_UID" and "BlackHole16ch_UID" is the answer and see whether repo maintainers have different opinion.
<img width="893" alt="image" src="https://user-images.githubusercontent.com/3164826/109593251-f6ea8a00-7b53-11eb-9e66-8b414e61ce40.png">

**Comment by devinroth** [comment]
AudioDeviceID are assigned by the OS. They change each time CoreAudio restarts. 

BlackHole is not a kext. It's a user space driver.

**Comment by devinroth** [comment]
You're right about the UIDs.

**Comment by ShikiSuen** [comment]
@devinroth Thanks for your response.

I just identified that SoundflowerBed uses device display names to identify whether the required drivers are installed.
I removed the brackets in the conditioning codes and now it shows up devices in the menu:
<img width="256" alt="image" src="https://user-images.githubusercontent.com/3164826/109593785-f8688200-7b54-11eb-8906-7cefb34fbb71.png">

However, it doesn't work.
I wonder whether SoundflowerBed does not know how to push its audio stream to user-space audio drivers.

I leave the forked project repo here: https://github.com/ShikiSuen/BlackHoleBed
to see whether anyone wants to troubleshoot it.
I have lacked experience of developing mac Apps, and I guess this is my stage to stop for a while.
